components/plugin-manager/api/src/commonMain/kotlin/PluginDescriptor.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.components.pluginmanager
+
+import kotlinx.serialization.Serializable
+
+/**
+ * A descriptor holding the metadata of a plugin.
+ */
+@Serializable
+data class PluginDescriptor(
```

Comment:
Question: Will there be a distinction between classes in the public API and internal data model classes and a mapping layer between those?

Reply:
I want to try to avoid having pure copies that to reduce the amount of boilerplate, but let's see how that works out.